core/src/cluster_info_vote_listener.rs

```diff
@@ -426,6 +431,15 @@ impl ClusterInfoVoteListener {
                         &verified_vote_packets,
                     )?;
                 }
+                // Check if we've crossed the feature boundary
```

Comment:
Reviewer note: Better to have this here a second time to avoid grabbing a read lock on `bank_forks` every iteration

Reply:
can we factor this optimization out to a separate PR to reduce the backport surface?

Reply:
> can we factor this optimization out to a separate PR to reduce the backport surface?

if you gimme write on your branch, i have this change made locally

Reply:
> can we factor this optimization out to a separate PR

To confirm, you're only talking about removing the `if` statement that skips reading feature set if we know the feature is already activated ? My initial comment was pointing out that we were reading in two places, but that was better to both 1) start with a valid value on first couple iterations of loop and 2) avoid spamming the lock.

That being said, `receive_and_process_vote_packets()` does `recv_timeout(200ms)` so my concern about having feature set happen every iteration of the loop is mitigated by `receive_and_process_vote_packets()` taking a nontrivial amount of time to return

Reply:
just gave you write access. if you're talking about the refactor to pass the flag rather than the feature set that was to avoid cloning or holding the bank forks lock for too long. 
the `if` statement could probably be removed taking into consideration the `200ms` sleep that steve pointed out.

Reply:
nope. the whole swap from `Option<FeatureSet>` to `bool`. https://github.com/solana-labs/solana/compare/master...t-nelson:solana:feature-flag-fix

we're already doing the inefficient thing and that inefficiency did not contribute to the testnet outage, so the optimization doesn't need to be backported

Reply:
sounds good, i'm fine with the clone.

Reply:
> sounds good, i'm fine with the clone.

hrmf, seems like github has some new default branch protection rules or something. it won't let me push my changes 'cause i rewrote your tip.  mind pulling my branch and pushing up the change?

ping me when we're readhy and i'll re-enable rebase+merge for this so we don't need a separate PR

Reply:
nice thanks, should now be 2 commits. hopefully github does the right thing and only backports the first one.

Reply:
> nice thanks, should now be 2 commits. hopefully github does the right thing and only backports the first one.

it'll try to do them both.  trivial to drop the last one tho